pkg/controller/podWatcher.go

```diff
@@ -209,7 +209,7 @@ func podEventsHandler(key string, indexer cache.Indexer) error {
 		// if we found 'ignore-update-event' annotation but the pod is in crash-loop-back
 		// we will still send the event so we can notify about it.
 		// in any other case we will send the event as long as 'ignore-update-event' annotations not set to true
-		if !ignoreEvent || onCrashLoopBack {
+		if watchEvent || onCrashLoopBack {
```

Comment:
this will not work well. you set the `watchEvent` variable to be false be default. this means we will not report any event (include Add/Delete) unless we are in update event and the user specifically said to watch the update event. this is of course not the logic we want